storage/client_raft_test.go

```diff
@@ -1102,3 +1104,61 @@ func TestRaftRemoveRace(t *testing.T) {
 		mtc.replicateRange(rangeID, 0, 2)
 	}
 }
+
+// TestStoreRangeRepair verifies that if a store becomes dead, the repair queue
+// will notice and remove any replicas on it.
+func TestStoreRangeRepair(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	mtc := &multiTestContext{}
+	mtc.timeUntilStoreDead = storage.TestTimeUntilStoreDead
+	mtc.Start(t, 3)
+	defer mtc.Stop()
+
+	sg := storeGossiper.NewStoreGossiper(mtc.gossip)
+
+	// Replicate the range to all stores.
+	replica := mtc.stores[0].LookupReplica(proto.KeyMin, nil)
+	mtc.replicateRange(replica.Desc().RangeID, 0, 1, 2)
+
+	// Initialize the gossip network.
+	var storeIDs []proto.StoreID
+	for _, s := range mtc.stores {
+		storeIDs = append(storeIDs, s.StoreID())
+	}
+	sg.GossipWithFunction(storeIDs, func() {
+		for _, s := range mtc.stores {
+			s.GossipStore()
+		}
+	})
+
+	aliveStoreIDs := []proto.StoreID{
+		mtc.stores[0].StoreID(),
+		mtc.stores[1].StoreID(),
+	}
+
+	rangeDesc := getRangeMetadata(proto.KeyMin, mtc, t)
+	if e, a := 3, len(rangeDesc.Replicas); e != a {
+		t.Fatalf("Expected %d replicas, only found %d. RangeDesc: %+v", e, a, rangeDesc)
+	}
+	maxTimeout := time.After(5 * time.Second)
+	for {
```

Comment:
OK. Can you add a comment? also: `for len(getRangeMetadata(proto.KeyMin, mtc, t)) != 2 {`?